### PR TITLE
Apply access controls to summary endpoint (#3298)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**
https://github.com/databiosphere/azul/issues/3298

Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passed in sandbox                               <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [ ] Started reindex in `dev`                              <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [ ] Checked for failures in `dev`                         <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [ ] Started reindex in `prod`                             <sub>or this PR does not require reindexing or does not target `prod`</sub>
- [ ] Checked for failures in `prod`                        <sub>or this PR does not require reindexing or does not target `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

**Comment by noah-aviel-dove** [comment]
Retracting review because I need to re-think the chain

**Review by danielsotirhos — APPROVED** [review]
LGTM

**Comment by noah-aviel-dove** [comment]
Re-requesting peer review since the upstream branch has changed and an integration test has been added.

File: test/integration_test.py [hunk 1]
```diff
@@ -784,27 +784,40 @@ def _source_ids_from_hits(hits: JSONs) -> Set[str]:
...
+            file_url = one(first(response['hits'])['files'])['url']
             response = self._http.request('GET', file_url)
             self.assertEqual(response.status, 404)
+        summary_url = furl(config.service_endpoint(),
+                           path='/index/summary',
+                           args={'catalog': catalog}).url
+
+        def _get_summary_file_count() -> int:
+            response = self._get_url_json(summary_url)
+            # FIXME: https://github.com/DataBiosphere/azul/issues/3318
```
Comment (danielsotirhos) [inline test/integration_test.py:810]:
```suggestion
            # FIXME: fileCount doesn't equal the sum of fileTypeSummaries[].count
                     https://github.com/DataBiosphere/azul/issues/3318
```

The FIXME should be placed where `fileCount` and/or `fileTypeSummaries[].count` are calculated.

**Review by danielsotirhos — CHANGES_REQUESTED** [review]
Look good. Just one nitpick about the fixme.

**Review by danielsotirhos — CHANGES_REQUESTED** [review]
A practice we started this year when adding FIXMEs is to either include the FIXME ticket number in the commit title, or better yet add the FIXME comment(s) in a separate commit with a title such as "Add FIXME (#3318)"

See: https://github.com/DataBiosphere/azul/pull/3210#pullrequestreview-703039596

```
(.venv) daniel@Crispin ~/repo/azul2 $ git log --pretty=format:"%h %cs %s" --date=short | grep FIXME
5968185f 2021-08-09 FIXME (#3291)
147cfa4f 2021-07-21 Add FIXME (#3152)
5c8e396d 2021-06-17 Add FIXME (#3071)
23cd201b 2021-05-25 FIXME: Consolidate base URL composition in tests (#3008)
a9428a1e 2021-05-10 Fix: FIXME references closed ticket (#2829, PR #3030)
dcae30ea 2021-05-10 Fix: FIXME references closed ticket (#2829)
9c23cb7f 2021-04-30 Add FIXME for prefix tooling (#3027)
4e5d4c37 2021-04-30 Add FIXME to rename SourceName (#2843)
...
```

File: test/integration_test.py [hunk 2]
```diff
@@ -814,22 +814,22 @@ def _source_ids_from_hits(hits: JSONs) -> Set[str]:
                        args={
                            'catalog': catalog,
                            'filters': json.dumps(source_filter)
-                       }).url
-            response = self._get_url_json(url)
+                       })
+            response = self._get_url_json(str(url))
```
Comment (danielsotirhos) [inline test/integration_test.py:818]: The change to `url` seems unnecessary for this PR and doesn't match how `summary_url` is composed and used soon after.

**Review by danielsotirhos — CHANGES_REQUESTED** [review]

**Comment by noah-aviel-dove** [comment]
When I rebased, the original fixup commit to adjust the requested changes disappeared during conflict resolution, since that part of the code needed to re-written anyway. The new fixup is unrelated, I just realized after pushing that #3318 had already been fixed.

**Review by danielsotirhos — APPROVED** [review]

**Comment by noah-aviel-dove** [comment]
The `fixup!` is to address a bug that very rarely caused an IT failure which I found after fixing the conflicts.

File: src/azul/service/source_controller.py [hunk 3]
```diff
@@ -63,3 +68,14 @@ def _add_implicit_sources_filter(self,
...
+
+    def _parse_filters(self,
+                       filters: Optional[Union[str, Filters]],
+                       source_ids: Optional[Set[str]] = None) -> MutableFilters:
+        if filters is None or isinstance(filters, str):
+            filters = super()._parse_filters(filters)
+        else:
+            filters = copy_json(filters)
+        if source_ids is not None:
+            self._add_implicit_sources_filter(filters, source_ids)
```
Comment (hannes-ucsc) [inline src/azul/service/source_controller.py:80]: This has nothing to do with parsing. The fact that the parsing of filters is done in close proximity to the application of the source filter is incidental and shouldn't lead to a conflation of the two operations.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: src/azul/service/repository_service.py [hunk 4]
```diff
@@ -151,8 +155,11 @@ def get_summary(self, catalog: CatalogName, filters: Filters):
 
         def transform_summary(entity_type):
             """Returns the key and value for a dict entry to transformation summary"""
+            entity_filters = (original_filters
```
[the remark this answers is not in the document]
Comment (noah-aviel-dove) [reply src/azul/service/repository_service.py:158]: Decided at parking lot to move `_add_implicit_sources_filter` to service layer (remove from `SourceController`)

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**